packages/world/src/WorldFactory.sol

```diff
@@ -25,15 +27,18 @@ contract WorldFactory is IWorldFactory {
     coreModule = _coreModule;
   }
 
-  /**
+  /*
    * @notice Deploys a new World instance, installs the CoreModule and transfers ownership to the caller.
    * @dev Uses the Create2 for deterministic deployment.
+   * @param _salt User defined salt for deterministic world addresses across chains
    * @return worldAddress The address of the newly deployed World contract.
    */
-  function deployWorld() public returns (address worldAddress) {
+  function deployWorld(bytes memory _salt) public returns (address worldAddress) {
     // Deploy a new World and increase the WorldCount
     bytes memory bytecode = type(World).creationCode;
-    uint256 salt = uint256(keccak256(abi.encode(msg.sender, worldCounts[msg.sender]++)));
+    uint256 salt = uint256(keccak256(abi.encode(msg.sender, _salt)));
+    worldCounts[msg.sender]++;
```

Comment:
Maybe we still need `worldCounts` now?

Reply:
I don't think we need world counts at all anymore?

Reply:
Would it be useful to store the deployed world addresses in an array? 

For example:
```solidity
mapping(address => address[]) public worldAddresses; 
```

This way anyone could see the number of deployed addresses via `worldAddresses.length` and would be able to get the respective addresses of the deployed worlds. 

However, if it is not needed, it would save gas to completely remove `worldCounts`

Reply:
We can remove it entirely, no need to maintain code or on-chain state if there isn't a proven use case